frontend/src/pages/modelRegistry/screens/RegisteredModels/usePrefillDeployModalFromModelRegistry.ts

```diff
@@ -0,0 +1,87 @@
+import { AlertVariant } from '@patternfly/react-core';
+import React from 'react';
+import { ProjectKind } from '~/k8sTypes';
+import useLabeledDataConnections from '~/pages/modelRegistry/screens/RegisteredModels/useLabeledDataConnections';
+import { RegisteredModelDeployInfo } from '~/pages/modelRegistry/screens/RegisteredModels/useRegisteredModelDeployInfo';
+import {
+  CreatingInferenceServiceObject,
+  InferenceServiceStorageType,
+  LabeledDataConnection,
+} from '~/pages/modelServing/screens/types';
+import { AwsKeys, EMPTY_AWS_SECRET_DATA } from '~/pages/projects/dataConnections/const';
+import useDataConnections from '~/pages/projects/screens/detail/data-connections/useDataConnections';
+import { DataConnection, UpdateObjectAtPropAndValue } from '~/pages/projects/types';
+
+const usePrefillDeployModalFromModelRegistry = (
+  projectContext: { currentProject: ProjectKind; dataConnections: DataConnection[] } | undefined,
+  createData: CreatingInferenceServiceObject,
+  setCreateData: UpdateObjectAtPropAndValue<CreatingInferenceServiceObject>,
+  registeredModelDeployInfo?: RegisteredModelDeployInfo,
+): [LabeledDataConnection[], boolean, Error | undefined] => {
+  const [fetchedDataConnections, dataConnectionsLoaded, dataConnectionsLoadError] =
+    useDataConnections(projectContext ? undefined : createData.project);
+  const allDataConnections = projectContext?.dataConnections || fetchedDataConnections;
+  const { dataConnections, storageFields } = useLabeledDataConnections(
+    registeredModelDeployInfo?.modelArtifactUri,
+    allDataConnections,
+  );
+
+  React.useEffect(() => {
+    if (registeredModelDeployInfo) {
+      setCreateData('name', registeredModelDeployInfo.modelName);
+      const recommendedDataConnections = dataConnections.filter(
+        (dataConnection) => dataConnection.isRecommended,
+      );
+
+      if (!storageFields) {
+        setCreateData('storage', {
+          awsData: EMPTY_AWS_SECRET_DATA,
+          dataConnection: '',
+          path: '',
+          type: InferenceServiceStorageType.EXISTING_STORAGE,
+        });
+      } else {
+        const prefilledAWSData = [
+          { key: AwsKeys.NAME, value: registeredModelDeployInfo.modelArtifactStorageKey || '' },
+          { key: AwsKeys.AWS_S3_BUCKET, value: storageFields.bucket },
+          { key: AwsKeys.S3_ENDPOINT, value: storageFields.endpoint },
+          { key: AwsKeys.DEFAULT_REGION, value: storageFields.region || '' },
+          ...EMPTY_AWS_SECRET_DATA,
+        ];
```

Comment:
Pre-filled all the fields if the URI is parsed correctly. Forgot to do that before, and now it's fixed.